### ISSUE TITLE
"Why can't some of the data in my training set be predicted with labels after training? I am working on instance segmentation."

### ISSUE THREAD

**Issue by zwf159** [ultralytics/ultralytics#18287]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

"Why can't some of the data in my training set be predicted with labels after training? I am working on instance segmentation."

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zwf159, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. This could include a small sample of your dataset, a snippet of your training code, and details about your model configuration. 

If this is a custom training ❓ Question, please provide as much detailed information as possible, such as:
- Dataset sample images (preferably with annotations)
- Details about your training settings (e.g., epochs, batch size, optimizer, etc.)
- Training logs or results you’ve observed
- Verification that you followed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 📈

Since you’re working on instance segmentation, make sure your data preparation adheres to the expected format outlined in the [Ultralytics Segmentation Guide](https://docs.ultralytics.com/tasks/segment/).

Join the vibrant Ultralytics community where it suits you best! For real-time chat or help, head over to [Discord](https://discord.com/invite/ultralytics) 🎧. For deeper discussions, check out our [Discourse](https://community.ultralytics.com). You can also share insights or ask questions in our growing [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please ensure you are using the latest version of the `ultralytics` package, as improvements and bug fixes are made frequently. Upgrade in a clean [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

Verify you have installed all the necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) as well.

## Environments

YOLO can also be run in a variety of verified environments. If your current environment might be causing this issue, explore one of the following options, which come preloaded with all dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

⚠️ Please note that this is an automated response to help you get started quickly. An Ultralytics engineer will review your query and provide further assistance as needed. 😊

**Comment by zwf159** [comment]
This is my training code, is it my data enhancement on too much?
from ultralytics import YOLO
import os
import torch

torch.cuda.device_count()
model = YOLO('yolo11x-seg.yaml').load('yolo11x-seg.pt') 
# model = YOLO(r'C:\Users\Administrator\Desktop\ultralytics-main\runs\detect\train\weights/best.pt')
if __name__ == '__main__':
    results = model.train(data='coco8-seg.yaml', mode='segment',epochs=1000, imgsz=640, batch=12, mixup=0.8,optimizer='SGD', copy_paste=0.8, translate=0.8,
                          shear=0.8, flipud=0.8, patience=300, fliplr=0.8, scale=0.8, device='0',workers=16,resume=True,amp=False)

    metrics = model.val()

**Comment by Y-T-G** [comment]
Are you training on your own dataset? `coco8-seg.yaml` is a toy dataset. It's only for demo.

**Comment by zwf159** [comment]
> Are you training on your own dataset? `coco8-seg.yaml` is a toy dataset. It's only for demo.

I modified it to use my own data.


# Ultralytics YOLO 🚀, AGPL-3.0 license
# COCO8-seg dataset (first 8 images from COCO train2017) by Ultralytics
# Documentation: https://docs.ultralytics.com/datasets/segment/coco8-seg/
# Example usage: yolo train data=coco8-seg.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── coco8-seg  ← downloads here (1 MB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
train: C:\Users\Administrator\Desktop\ultralytics-main\tianke\images\train # train images (relative to 'path') 4 images
val: C:\Users\Administrator\Desktop\ultralytics-main\tianke\images\val # val images (relative to 'path') 4 images
test: # test images (optional)

# Classes
names:
  0: edge
  
"I think, theoretically, there shouldn't be a situation where the training data used for segmentation has very low confidence in some parts."

**Comment by Y-T-G** [comment]
How many images do you have?

**Comment by zwf159** [comment]
> How many images do you have?

700 images are divided into an 8:2 ratio for training and validation. I just tested the yolo11l-seg model and found its performance to be better than x. I'm not sure if it's due to a bug in x or something else. Here are two comparison images: the one with a mask is from model l, and the one without is from model x. I suspect it's a bug because the confidence values of model l are relatively high. Logically, model x shouldn't fail at segmentation. My confidence threshold is set to 0.2.

![TKT91EJH03_1_3](https://github.com/user-attachments/assets/263f38db-bfc5-4d7c-a3a2-1225e49419ce)
![TKT91EJH03_1_3](https://github.com/user-attachments/assets/411a43a0-5ccd-49b9-aa94-114e6d1841fe)

**Comment by Y-T-G** [comment]
Your objects are too large. You should use smaller imgsz. Like 320.

**Comment by zwf159** [comment]
> Your objects are too large. You should use smaller imgsz. Like 320.

Do you mean that the images I uploaded are too large and need to be resized to 320*320 for you, or that the training images need to be resized to 320? Currently, they are 640.

**Comment by Y-T-G** [comment]
The objects in your training image cover most of the image. They're too large. So you need to use smaller imgsz during training

**Comment by zwf159** [comment]
> The objects in your training image cover most of the image. They're too large. So you need to use smaller imgsz during training
好的，但是我有一些情况，是比较小的情况，如果这时候减少图片尺寸的话会不会造成小情况的识别效果不好，比如这张，或许我增加小目标检测头？
![TKZG07FEB03_5_1](https://github.com/user-attachments/assets/fce05bc7-1aa1-4e78-b7af-5f9721e44bb2)

**Comment by glenn-jocher** [comment]
Reducing `imgsz` could indeed impact the detection of smaller objects, as resolution affects feature extraction. Instead of solely reducing `imgsz`, consider using a multi-scale training approach or augmenting your dataset with a mix of crops and scales. Adding a detection head for small objects is an advanced option but can be effective if implemented correctly.
